doc/user/content/integrations/upstash-kafka.md

```diff
@@ -0,0 +1,114 @@
+
+    If you already have a topic created, you can skip this step.
+
+    Otherwise, you can follow the steps [here](https://docs.upstash.com/kafka#create-a-topic) to create a topic.
+
+    Once you have a topic, you can produce a message to it by clicking on the topic name and then clicking on the **Messages** tab, and then clicking on the **Produce** button.
+
+4. #### Create a source in Materialize
+    a. Open the [Upstash console](https://console.upstash.com/login) and select your cluster
+
+    b. Click on **Details**
+
+    c. Copy the URL under **Endpoint**. This will be your `<upstash-broker-url>` going forward
+
+    d. From the _psql_ terminal, run the following command. Replace `<upstash_kafka>` with whatever you want to name your source. The broker URL is what you copied in step c of this subsection. The `<topic-name>` is the name of the topic you created in Step 3. The `<your-username>` and `<your-password>` are from the _Create new credentials_ step.
+
+    ```sql
+      CREATE SECRET upstash_username AS '<your-username>';
+      CREATE SECRET upstash_password AS '<your-password>';
+
+      CREATE CONNECTION <upstash_kafka> TO KAFKA (
+          BROKER '<upstash-broker-url>',
+          SASL MECHANISMS = 'SCRAM-SHA-256',
+          SASL USERNAME = SECRET upstash_username,
+          SASL PASSWORD = SECRET upstash_password
+      );
+
+      CREATE SOURCE <source-name>
+        FROM KAFKA CONNECTION upstash_kafka (TOPIC '<topic-name>')
+        FORMAT BYTES
+        WITH (SIZE = '3xsmall');
+    ```
+
+    e. If the command executes without an error and outputs _CREATE SOURCE_, it means that you have successfully connected Materialize to your Upstash Kafka cluster. You can quickly test your connection by running the following statement:
+    ```sql
+      SELECT convert_from(data, 'utf8') from <source-name>;
+    ```
+
+    **Note:** The example above walked through creating a source, which is a way of connecting Materialize to an external data source. We created a connection to Upstash Kafka using SASL authentication, using credentials securely stored as secrets in Materialize's secret management system. For input formats, we used `bytes`, however, Materialize supports various other options as well. For example, you can ingest Upstash Kafka messages formatted in [JSON, Avro, and Protobuf](/sql/create-source/kafka/#supported-formats). You can find more details about the various different supported formats and possible configurations [here](/sql/create-source/kafka/).
+
+5. #### Create a sink in Materialize
+
+    A [sink](/sql/create-sink) is a way to write data from Materialize to an external data source like a Kafka cluster.
+
+    To create a sink, from the _psql_ terminal, run the following command:
+
+    ```sql
+      CREATE SINK <sink-name>
+        FROM <source, table or mview>
+        INTO KAFKA CONNECTION <upstash_kafka> (TOPIC '<sink-topic-name>')
+        FORMAT JSON
+        ENVELOPE DEBEZIUM
```

Comment:
Is it worth explaining why to use a Debezium envelope vs upserts. It threw me for a second to see Debezium (and didn't realize that was a valid envelope for sinks)

Reply:
Good point, just added a sentence to explain that! Let me know what you think.